### PR TITLE
Split client into client and transport (preelminary PR for Dark Sky integration)

### PR DESCRIPTION
**Pull request by asoltysik**
Due to moving and splitting files around, git diff was not too kind to us, hence this PR. Could not make it smaller though.

Most important changes:

- constructing the clients is now only possible through functions in `providers.openweather.OpenWeatherMap` object, constructors of these clients are package-private. This is also important because of possible changes comming with #64
- API-agnostic `Transport` was split from the `Client`. It's similar to transport in 0.3, but contains everything needed for decoding (where transport in 0.3 returned jsons)
- Timeout functionality from the caching client was also moved out to `TimeoutHttpTransport`

### CODE REVIEW

File: src/main/scala/com.snowplowanalytics/weather/providers/openweather/OpenWeatherMap.scala [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import Errors.{InvalidConfigurationError, WeatherError}
+import Responses.History
+
+object OpenWeatherMap {
+
+  /**
+   * Create `OwmClient` with specified underlying `Transport`
+   * @param transport instance of `Transport` which will do the actual sending of data
+   */
+  def basicClient[F[_]: Sync](transport: Transport[F]): OwmClient[F] =
```
Comment (BenFradet) [inline src/main/scala/com.snowplowanalytics/weather/providers/openweather/OpenWeatherMap.scala]: I don't think we should expose this one?

File: src/main/scala/com.snowplowanalytics/weather/providers/openweather/OpenWeatherMap.scala [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+   * @param cacheSize amount of history requests storing in cache
+   *                  it's better to store whole OWM packet (5000/50000/150000)
+   *                  plus some space for errors (~1%)
+   * @param geoPrecision nth part of 1 to which latitude and longitude will be rounded
+   *                     stored in cache. For eg. coordinate 45.678 will be rounded to
+   *                     values 46.0, 45.5, 45.7, 45.78 by geoPrecision 1,2,10,100 respectively
+   *                     geoPrecision 1 will give ~60km infelicity if worst case; 2 ~30km etc
+   * @param transport instance of `Transport` which will do the actual sending of data
+   */
+  def cacheClient[F[_]: Concurrent](cacheSize: Int, geoPrecision: Int, transport: Transport[F]): F[OwmCacheClient[F]] =
```
Comment (BenFradet) [inline src/main/scala/com.snowplowanalytics/weather/providers/openweather/OpenWeatherMap.scala]: same here?

File: src/main/scala/com.snowplowanalytics/weather/providers/openweather/OwmClient.scala [hunk 3]
```diff
@@ -13,75 +13,171 @@
...
 /**
- * Asynchronous OpenWeatherMap client
+ * Non-caching OpenWeatherMap client
  *
- * @param appId API key
+ * @tparam F effect type
  */
-case class OwmClient[F[_]: Sync](appId: String, apiHost: String = "api.openweathermap.org", ssl: Boolean = false)
-    extends Client[F] {
+class OwmClient[F[_]] private[openweather] (private[openweather] val transport: Transport[F]) {
```
Comment (BenFradet) [inline src/main/scala/com.snowplowanalytics/weather/providers/openweather/OwmClient.scala]: why is `transport` `private[openweather]` and not just simply `private`?

**Comment by asoltysik** [comment]
@BenFradet done

**Pull request merged**